File: Modules/module_jetpack/src/main/java/com/example/william/my/module/jetpack/adapter/PagingAdapter.java
package com.example.william.my.module.jetpack.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.paging.PagingDataAdapter;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.william.my.module.bean.ArticleDetailBean;
import com.example.william.my.module.jetpack.R;

import org.jetbrains.annotations.NotNull;

public class PagingAdapter extends PagingDataAdapter<ArticleDetailBean, RecyclerView.ViewHolder> {

    public PagingAdapter(@NotNull DiffUtil.ItemCallback<ArticleDetailBean> diffCallback) {
        super(diffCallback);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.jet_item_recycle, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        ArticleDetailBean articleDetailBean = getItem(position);
        ((ViewHolder) holder).mTextView.setText(articleDetailBean == null ? "" : position + ". " + articleDetailBean.getTitle());
    }

    private static class ViewHolder extends RecyclerView.ViewHolder {

        public final TextView mTextView;

        public ViewHolder(View itemView) {
            super(itemView);
            this.mTextView = itemView.findViewById(R.id.item_recycle_textView);
        }
    }
}

File: modules/module_sample/src/main/kotlin/com/example/william/my/module/sample/paging/PagingAdapter.kt
package com.example.william.my.module.sample.paging

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.example.william.my.bean.data.ArticleDetailBean
import com.example.william.my.module.sample.databinding.SampleItemRecyclerBinding

/**
 * Paging RecyclerView 适配器
 */
class PagingAdapter(diffCallback: DiffUtil.ItemCallback<ArticleDetailBean>) :
    PagingDataAdapter<ArticleDetailBean, PagingAdapter.ViewHolder>(diffCallback) {

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = position.toString() + ". " + getItem(position)?.title
        holder.itemBind.itemTextView.text = item
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val bind = SampleItemRecyclerBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(bind)
    }

    class ViewHolder(bind: SampleItemRecyclerBinding) : RecyclerView.ViewHolder(bind.root) {
        var itemBind: SampleItemRecyclerBinding = bind
    }
}